### PR TITLE
Docs correction. Installation and Run Federation sections

### PR DESCRIPTION
**Pull request by dmitryagapov**
running federation director flow update

### CODE REVIEW

File: docs/install.rst [hunk 1]
```diff
@@ -27,21 +27,21 @@ Install the Package
 Follow this procedure to prepare the environment and install the |productName| package.
 Perform this procedure on every node in the federation.
 
-1. Install a Python\* \  3.6 (or higher) virtual environment. 
+1. Install a Python 3.8 (or higher) virtual environment using venv.
```
Comment (aleksandr-mokrov) [inline docs/install.rst]: python_requires='>=3.6, <3.9'

File: docs/running_the_federation.rst [hunk 2]
```diff
@@ -9,13 +9,207 @@ Run the Federation
...
+:doc:`source/workflow/running_the_federation.agg_based`
+    With this workflow, the federation is terminated when the experiment is finished.
+
+.. # Copyright (C) 2020-2021 Intel Corporation
+.. # SPDX-License-Identifier: Apache-2.0
+
+.. _director_workflow:
+
+************************
+Director-Based Workflow
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst:22]: Duplication? 
image

File: docs/running_the_federation.rst [hunk 3]
```diff
@@ -9,13 +9,207 @@ Run the Federation
...
+Follow the procedure in the director-based workflow to become familiar with the setup required and APIs provided for each role in the federation: *Director manager*, *Collaborator manager*, and *Experiment manager (data scientist)*.
+
+.. note::
+    The Open Federated Learning (|productName|) interactive Python API enables the Experiment manager (data scientists) to define and start a federated learning experiment from a single entry point: a Jupyter\*\  notebook or a Python\*\  script.
+
+    See :doc:`director_based_workflow.interactive_api` for details.
+
+An overview of this workflow is shown below.
+
+.. figure:: ../openfl/static_diagram.svg
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]: The picture is not displayed
image

File: docs/running_the_federation.rst [hunk 4]
```diff
@@ -9,13 +9,207 @@ Run the Federation
...
+.. centered:: Overview of the Director-Based Workflow
+
+
+.. # Copyright (C) 2020-2021 Intel Corporation
+.. # SPDX-License-Identifier: Apache-2.0
+
+
+.. _establishing_federation_director:
+
+Director Manager: Set Up the Director
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst:61]: Also duplicated section with `openfl/docs/source/workflow/director_based_workflow.roles.rst` file, why not just change this file?

Comment (dmitryagapov) [reply docs/running_the_federation.rst:61]: Could you please review the content. This duplicate can help you to see the difference between versions. Before merge all of them will be deleted

File: docs/install.rst [hunk 5]
```diff
@@ -27,21 +27,21 @@ Install the Package
...
 
-      python3 -m virtualenv env_name
+      python3 -m venv venv
 
 3. Activate the virtual environment.
 
    .. code-block:: console
 
-      source env_name/bin/activate
+      source venv/bin/activate
```
Comment (igor-davidyuk) [inline docs/install.rst:44]: It is better to leave the `env_name` parameter in my opinion

File: docs/running_the_federation.rst [hunk 6]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+    - :ref:`establishing_federation_director`
+    - :ref:`establishing_federation_envoy`
+    - :ref:`establishing_federation_experiment_manager`
+
+Follow the procedure in the director-based workflow to become familiar with the setup required and APIs provided for each role in the federation: *Director manager*, *Collaborator manager*, and *Experiment manager (data scientist)*.
+
+.. note::
+    The Open Federated Learning (|productName|) interactive Python API enables the Experiment manager (data scientists) to define and start a federated learning experiment from a single entry point: a Jupyter\*\  notebook or a Python\*\  script.
+
+    See :doc:`director_based_workflow.interactive_api` for details.
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]:
```
WARNING: unknown document: director_based_workflow.interactive_api
```
image

File: docs/running_the_federation.rst [hunk 7]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+
+.. _establishing_federation_director:
+
+Director Manager: Set Up the Director
+=====================================
+
+The *Director manager* sets up the *Director*, which is the central node of the federation.
+
+    - :ref:`optional_step_create_pki_using_step_ca`
+    - :ref:`step0_install_director_prerequisites`
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]: STEP 0,  STEP 1 looks strange
image
May be it is better to leave old version? (STEP 1, STEP 2, STEP 3), or just OPTIONAL,  STEP 1 STEP 2

File: docs/running_the_federation.rst [hunk 8]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+
+
+You can use the `"Hello Federation" bash script <https://github.com/intel/openfl/blob/develop/tests/github/test_hello_federation.sh>`_ to quickly create a federation (an aggregator node and two collaborator nodes) to test the project pipeline.
+
+.. literalinclude:: ../tests/github/test_hello_federation.sh
+  :language: bash
+
+However, continue with the following procedure for details in creating a federation with an aggregator-based workflow.
+
+    :doc:`STEP 1: Create a Workspace on the Aggregator <running_the_federation.baremetal>`
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]:
```
WARNING: unknown document: running_the_federation.baremetal
```
May be we should remove `<running_the_federation.baremetal>`?

File: docs/running_the_federation.rst [hunk 9]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+
+        - Creates a federated learning workspace on one of the nodes.
+
+
+    :doc:`STEP 2: Configure the Federation <running_the_federation.certificates>`
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]:
```
WARNING: unknown document: running_the_federation.certificates
```

File: docs/running_the_federation.rst [hunk 10]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+
+        - Ensures each node in the federation has a valid public key infrastructure (PKI) certificate.
+        - Distributes the workspace from the aggregator node to the other collaborator nodes.
+
+
+    :doc:`STEP 3: Start the Federation <running_the_federation.start_nodes>`
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]:
```
 WARNING: unknown document: running_the_federation.start_nodes
```

File: docs/running_the_federation.rst [hunk 11]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+
+.. _running_the_federation_docker_base_image:
+
+Option 1: Deploy a Federation in a Docker Container
+===================================================
+
+Prerequisites
+-------------
+
+You have already built an |productName| image. See :doc:`../../install.docker` for details.
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]:
```
WARNING: unknown document: ../../install.docker
```
image

File: docs/running_the_federation.rst [hunk 12]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+            bash
+
+
+    .. note::
+
+        The FL plan should be initialized with the FQDN of the node where the aggregator container will be running.
+
+3. Generate public key infrastructure (PKI) certificates for all collaborators and the aggregator. See :doc:`/source/utilities/pki` for details.
+
+4. :doc:`Start the federation <running_the_federation.start_nodes>`.
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]:
```
WARNING: unknown document: running_the_federation.start_nodes
```
This link does not work, 
```suggestion
4. :ref:`Start the federation <running_the_federation.start_nodes>`.
```
May be this suggestion should work correctly

File: docs/running_the_federation.rst [hunk 13]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+.. _step1_start_the_director:
+
+STEP 1: Start the Director
+--------------------------
+
+Start the Director on a node with at least two open ports. See :ref:`openfl_ll_components` to learn more about the Director entity.
+
+1. Create a Director workspace with a default config file.
+
+    .. code-block:: console
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst:89]: Please, add blank line due to 
```
openfl/docs/running_the_federation.rst:91: WARNING: Error in "code-block" directive:
maximum 1 argument(s) allowed, 6 supplied.

.. code-block:: console
    fx director create-workspace -p path/to/director_workspace_dir
```

File: docs/running_the_federation.rst [hunk 14]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+
+2. Modify the Director config file according to your federation setup.
+
+ The default config file contains the Director node FQDN, an open port, path of certificates, and :code:`sample_shape` and :code:`target_shape` fields with string representation of the unified data interface in the federation.
+
+3. Start the Director.
+
+ If mTLS protection is not set up, run this command.
+
+    .. code-block:: console
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst:103]: Please, add blank line

btw, for now it looks like:
image

File: docs/running_the_federation.rst [hunk 15]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+       fx director start --disable-tls -c director_config.yaml
+
+ If you have a federation with PKI certificates, run this command.
+
+    .. code-block:: console
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst:109]: Please, add blank line

File: docs/running_the_federation.rst [hunk 16]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+
+
+.. _step1_start_the_envoy:
+
+STEP 3: Start the Envoy
+-----------------------
+
+1. Create an Envoy workspace with a default config file and shard descriptor Python\*\  script.
+
+    .. code-block:: console
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst:152]: Please, add blank line

File: docs/running_the_federation.rst [hunk 17]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+        :code:`get_dataset()` method accepts the dataset_type (for instance train, validation, query, gallery) and returns
+        an iterable object with samples and targets.
+
+        MyShardDescriptor should be inherent from :code:`openfl.interface.interactive_api.shard_descriptor.ShardDescriptor`. It should implements :code:`get_dataset`, :code:`sample_shape` and :code:`target_shape` methods o describe the way data samples and labels will be loaded from disk during training.
+
+3. Start the Envoy.
+
+ If mTLS protection is not set up, run this command.
+
+    .. code-block:: console
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst:176]: Please, add blank line

File: docs/running_the_federation.rst [hunk 18]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+        fx envoy start \
+            -n "$ENVOY_NAME" \
+            --disable-tls \
+            --envoy-config-path envoy_config.yaml \
+            -dh director_fqdn \
+            -dp port
+
+ If you have a federation with PKI certificates, run this command.
+
+    .. code-block:: console
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst:189]: Please, add blank line

File: docs/running_the_federation.rst [hunk 19]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+    .. note::
+        The shard descriptor is an object to provide a unified data interface for FL experiments.
+        The shard descriptor implements :code:`get_dataset()` method as well as several additional
+        methods to access **sample shape**, **target shape**, and **shard description** that may be used to identify
+        participants during experiment definition and execution.
+
+        :code:`get_dataset()` method accepts the dataset_type (for instance train, validation, query, gallery) and returns
+        an iterable object with samples and targets.
+
+        MyShardDescriptor should be inherent from :code:`openfl.interface.interactive_api.shard_descriptor.ShardDescriptor`. It should implements :code:`get_dataset`, :code:`sample_shape` and :code:`target_shape` methods o describe the way data samples and labels will be loaded from disk during training.
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]:
```suggestion
        MyShardDescriptor should be inherited from :code:`openfl.interface.interactive_api.shard_descriptor.ShardDescriptor`. It should implement :code:`get_dataset`, :code:`sample_shape` and :code:`target_shape` methods to describe the way data samples and labels will be loaded from disk during training.
```

File: docs/running_the_federation.rst [hunk 20]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+
+.. _establishing_federation_experiment_manager:
+
+Experiment Manager: Describe an Experiment
+==========================================
+
+The process of defining an experiment is decoupled from the process of establishing a federation.
+The Experiment manager (or data scientist) is able to prepare an experiment in a Python environment.
+Then the Experiment manager registers experiments into the federation using :ref:`Interactive Python API <interactive_api>`
+that is allow to communicate with Director using a gRPC client.
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]:
```suggestion
that is allow to communicate with the Director using a gRPC client.
```

File: docs/running_the_federation.rst [hunk 21]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+
+There are two ways to run federation without Director:
+
+- :ref:`Bare metal approach <running_the_federation_manual>`
+- :ref:`Docker approach <running_the_federation_docker>`
+
+
+This workflow uses short-lived components in a federation, which is terminated when the experiment is finished. The components are as follows:
+
+- The *Collaborator* uses a local dataset to train a global model and the *Aggregator* receives model updates from *Collaborators* and aggregate them to create the new global model.
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]:
```suggestion
- The *Collaborator* uses a local dataset to train a global model and the *Aggregator* receives model updates from *Collaborators* and aggregates them to create the new global model.
```

File: docs/running_the_federation.rst [hunk 22]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+.. note::
+    Use the Federated Learning plan (FL plan) to modify the federation workspace to your requirements in an **aggregator-based workflow**.
+
+
+The FL plan is described by the **plan.yaml** file located in the **plan** directory of the workspace.
+
+
+Each YAML top-level section contains the following subsections:
+
+- ``template``: The name of the class including top-level packages names. An instance of this class is created when plan gets initialized.
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]:
```suggestion
- ``template``: The name of the class including top-level packages names. An instance of this class is created when the plan gets initialized.
```

File: docs/running_the_federation.rst [hunk 23]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+- ``settings``: The arguments that are passed to the class constructor.
+- ``defaults``: The file that contains default settings for this subsection.
+  Any setting from defaults file can be overriden in the **plan.yaml** file.
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]:
```suggestion
  Any setting from the default file can be overridden in the **plan.yaml** file.
```

File: docs/running_the_federation.rst [hunk 24]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+
+
+    .. code-block:: console
+
+		fx plan initialize
+
+
+    This command initializes the FL plan and autopopulates the `fully qualified domain name (FQDN) <https://en.wikipedia.org/wiki/Fully_qualified_domain_name>`_ of the aggregator node. This FQDN is embedded within the FL plan so the collaborator nodes know the address of the externally accessible aggregator server to connect to.
+
+    If you have connection issues with the autopopulated FQDN in the FL plan, you can do **one of the following**:
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]:
```suggestion
    If you have connection issues with the auto populated FQDN in the FL plan, you can do **one of the following**:
```

Comment (alexey-khorkin) [inline docs/running_the_federation.rst]:
```suggestion
    If you have connection issues with the auto populated FQDN in the FL plan, you can do **one of the following**:
```

File: docs/running_the_federation.rst [hunk 25]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+
+        The protobuf file with the initial weights is found in **${WORKSPACE_TEMPLATE}_init.pbuf**.
+
+
+    .. code-block:: console
+
+		fx plan initialize
+
+
+    This command initializes the FL plan and autopopulates the `fully qualified domain name (FQDN) <https://en.wikipedia.org/wiki/Fully_qualified_domain_name>`_ of the aggregator node. This FQDN is embedded within the FL plan so the collaborator nodes know the address of the externally accessible aggregator server to connect to.
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]:
```suggestion
    This command initializes the FL plan and auto populates the `fully qualified domain name (FQDN) <https://en.wikipedia.org/wiki/Fully_qualified_domain_name>`_ of the aggregator node. This FQDN is embedded within the FL plan so the collaborator nodes know the address of the externally accessible aggregator server to connect to.
```

File: docs/running_the_federation.rst [hunk 26]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+
+    If you have connection issues with the autopopulated FQDN in the FL plan, you can do **one of the following**:
+
+	- OPTION 1: override the autopopulated FQDN value with the :code:`-a` flag.
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst]:
```suggestion
	- OPTION 1: override the auto populated FQDN value with the :code:`-a` flag.
```

File: docs/running_the_federation.rst [hunk 27]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+
+STEP 0: Install |productName|
+-----------------------------
+
+Install |productName| in a Python\*\ virtual environment. See :ref:`install_package` for details.
+
+
+.. _step1_start_the_envoy:
+
+STEP 3: Start the Envoy
```
Comment (aleksandr-mokrov) [inline docs/running_the_federation.rst]: STEP 1. Or STEP 0 -> STEP 1 and this one should be second.

File: docs/running_the_federation.rst [hunk 28]
```diff
@@ -9,13 +9,737 @@ Run the Federation
...
+-----
+
+Each task subsection contains the following:
+
+- ``function``: The function name to call.
+  The function must be the one defined in :class:`TaskRunner <openfl.federated.TaskRunner>` class.
+- ``kwargs``: kwargs passed to the ``function``.
+
+.. note::
+    See an `example <https://github.com/intel/openfl/blob/develop/openfl/federated/task/runner.py>`_ of the :class:`TaskRunner <openfl.federated.TaskRunner>` class for details.
```
Comment (alexey-khorkin) [inline docs/running_the_federation.rst:606]: May be we should use `CoreTaskRunner` example here for more details? https://github.com/intel/openfl/blob/develop/openfl/federated/task/task_runner.py

File: docs/install.rst (hunk 5 above)
Comment (dmitryagapov) [reply docs/install.rst:44]: I'm not sure. This command could be copied without any modifications. But if we change it to `env_name`, user should define his own name every time.

File: docs/running_the_federation.rst [hunk 29]
```diff
@@ -9,13 +9,732 @@ Run the Federation
...
+    .. note::
+        The shard descriptor is an object to provide a unified data interface for FL experiments.
+        The shard descriptor implements :code:`get_dataset()` method as well as several additional
+        methods to access **sample shape**, **target shape**, and **shard description** that may be used to identify
+        participants during experiment definition and execution.
+
+        :code:`get_dataset()` method accepts the dataset_type (for instance train, validation, query, gallery) and returns
+        an iterable object with samples and targets.
+
+        MyShardDescriptor should be inherent from :code:`openfl.interface.interactive_api.shard_descriptor.ShardDescriptor`. It should implements :code:`get_dataset`, :code:`sample_shape` and :code:`target_shape` methods o describe the way data samples and labels will be loaded from disk during training.
```
Comment (aleksandr-mokrov) [inline docs/running_the_federation.rst]:
```suggestion
        User shard descriptor should be inherent from :code:`openfl.interface.interactive_api.shard_descriptor.ShardDescriptor`. It should implements :code:`get_dataset`, :code:`sample_shape` and :code:`target_shape` methods o describe the way data samples and labels will be loaded from disk during training.
```

**Review by alexey-khorkin — CHANGES_REQUESTED** [review]

File: docs/running_the_federation.rst (hunk 8 above)
Comment (aleksandr-mokrov) [reply docs/running_the_federation.rst]: It should be source/workflow/running_the_federation.baremetal and for others running_the_federation.* links

**Review by aleksandr-mokrov — APPROVED** [review]

**Review by ViktoriiaRomanova — APPROVED** [review]

**Review by alexey-gruzdev — APPROVED** [review]

**Pull request merged**